README.md
# mt-demo README

This is the README for your extension "mt-demo". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: enable/disable this extension
* `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

-----------------------------------------------------------------------------------------------------------
## Following extension guidelines

Ensure that you've read through the extensions guidelines and follow the best practices for creating your extension.

* [Extension Guidelines](https://code.visualstudio.com/api/references/extension-guidelines)

## Working with Markdown

**Note:** You can author your README using Visual Studio Code.  Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux)
* Toggle preview (`Shift+CMD+V` on macOS or `Shift+Ctrl+V` on Windows and Linux)
* Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (macOS) to see a list of Markdown snippets

### For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**


CHANGELOG.md
# Change Log

All notable changes to the "mt-demo" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

src/test.ts
import fetch from "./fetch";

export function test() {
    fetch.post('123', {}, {});
}

src/fetch.ts
// 将axios封装为fetch
import axios from "axios";

function checkStatus(response: any) {
  // 隐身模式首次打开问题
  if (!response) {
    return;
  }
  if (
    response.status === 200 ||
    response.status === 201 ||
    response.status === 304 ||
    response.status === 404
  ) {
    return response;
  }
  let message: string | null = null;
  switch (response.status) {
    case 404:
      message = "访问路径不存在";
      break;
    case 500:
      message = "服务器错误";
      break;
    default:
      message = null;
  }
  return {
    code: response.status,
    message: message || response.statusText,
    data: response.statusText,
  };
}

function checkCode(response: any) {
  const res = response && response.data;
  const { code, data } = res;
  return res;
}

export default {
  post(url: string, headers: any, data: any) {
    return axios({
      method: "post",
      url,
      data: data,
      timeout: 60000,
      headers: headers,
    })
      .then(checkStatus)
      .then(checkCode);
  },
  get(url: string, headers: any, data?: any) {
    return axios({
      method: "get",
      url,
      params: data && data.params,
      timeout: 60000,
      headers: headers,
    })
      .then(checkStatus)
      .then(checkCode);
  },
  put(url: string, headers: any, data: any) {
    return axios({
      method: 'put',
      url,
      data: data,
      timeout: 60000,
      headers: headers,
    })
      .then(checkStatus)
      .then(checkCode);
  }
};


src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import * as sidebar from './nodeDependencies';

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
	
	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
	console.log('Congratulations, your extension "mt-demo" is now active!');

	// The command has been defined in the package.json file
	// Now provide the implementation of the command with registerCommand
	// The commandId parameter must match the command field in package.json
	let disposable = vscode.commands.registerCommand('mt-demo.helloWorld', () => {
		// The code you place here will be executed every time your command is executed
		// Display a message box to the user
		vscode.window.showInformationMessage('Hello World from mt-demo!');
	});

	// vscode.window.createTreeView('requirement-list', {
	// 	treeDataProvider: new NodeDependenciesProvider()
	// });

	const sidebar_test = new sidebar.EntryList();
    vscode.window.registerTreeDataProvider("requirement-list",sidebar_test);
	vscode.commands.registerCommand("requirement-list.openChild",args => {
        vscode.window.showInformationMessage(args);
    });

	context.subscriptions.push(disposable);
}

// this method is called when your extension is deactivated
export function deactivate() {}


src/nodeDependencies.ts
import { time } from 'console';
import { hasMagic } from 'glob';
// import { request, get } from 'http';
import { getHeapStatistics } from 'v8';
// import { emitWarning } from 'process';
import * as vscode from 'vscode';
// import { url } from 'inspector';
import fetch from './fetch';

var rp = require('request-promise');

import request  from 'sync-request';
import axios from 'axios';


// 树节点
export class EntryItem extends vscode.TreeItem
{
}

//树的内容组织管理
export class EntryList implements vscode.TreeDataProvider<EntryItem>
{
    res_str: string = '';
    rootItems: EntryItem[] = [];
    onDidChangeTreeData?: vscode.Event<void | EntryItem | null | undefined> | undefined;
    getTreeItem(element: EntryItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
        return element;
    }
    getChildren(element?: EntryItem): vscode.ProviderResult<EntryItem[]> {
        if (element) {//子节点
            var childs = [];
            for (let index = 0; index < 3; index++) {
                let str = index.toString();
                var item = new EntryItem(str,vscode.TreeItemCollapsibleState.None);
                item.command = {command:"requirement-list.openChild", //命令id
                                title:"标题",
                                arguments:[str] //命令接收的参数
                                };
                childs[index] = item;
                console.log(str);
            }
            return childs;
        } else { //根节点
            console.log('root enter');

            
            // this.post();
            // this.post();

            // this.test();
            // let p = this.rp_test();
       
           

            // var res = request('POST', url, {
            // json: {username: 'ForbesLindesay'},
            // });
            // var user = JSON.parse(res.getBody('utf8'));


            // console.log("this.res_str");
            // console.log(this.res_str);
            
            
            // let bbb = JSON.parse(this.res_str);
            
            // let ret: EntryItem[] = [];
            //         for (let item of bbb.data.items) {
            //             ret.push(new EntryItem(item.name, vscode.TreeItemCollapsibleState.None));
            //         }
            // console.log('return');
            // console.log(ret);

            // this.sync_req()
            // var res = request('POST', 'https://www.baidu.com');
            // console.log(res.getBody());

            this.post();
            return [];
            // return [new EntryItem("123", vscode.TreeItemCollapsibleState.None)];
        }
    }

    sync_req() {
        // feature test
        const postData = JSON.stringify({
            "query":[
                {
                    "field":"state",
                    "type":"TERMS",       
                    "valueList":[
                        "规划中",
                        "开发中",
                        "交互视觉设计",
                        "待开发",
                        "需求评审222",
                        "技术评审3333",
                        "test",
                        "需求设计中"
                    ]
                },
                {
                    "field":"assigned",
                    "type":"TERMS",
                    "valueList":[
                        "muzhantong"
                    ]
                }
            ],
            "displayFieldList" : [
                "id", "name", "projectId", "state"]            
        });

        let headers = {
            "ssoClientId":"d2a0562363",
            "ssoId":"eAGFji9IA1EcgHltXJIDwXhxWzh-7_Z7f2bytjmMA4NgkXfv_Q4M3oIbFoMz2PzDTC4ZRJMboqBZg2brwHa6YrBoExUx27_v4yuwqY-HAQvG78P8DiJ_jVY7XZOFRGE7o_UwpdkAHOhIcMGVE4joEiFU2TjuDE94CrI2Zn5piZJFSxlt1hu1WFcbMQjeFHWcj5VqYgW00HWMqjwK3vb6Z_dQZNG_Yf2zNuctPH2eP99C6_H18OISthn3vNjadjfrtDacPz25Os33r_OD3Xy083LcmwxPJqPeDAu2bo5KxV-nzwp_fwMWWomawAhpEmvSqlRkJSUuQqVNmmq-wqWsgAQFKDQsBwRICaYVgrJG4NrgN0vOkjROc3JfwZpvug**eAEFwQkRwEAIBDBLhYE9kMOz-JfQhN5x4-4TcYYjDimKlWfHIdQtv5dLlGqMNTpvrajiZfMDSUASBg**pJV6fyj_WZa4Z_4lYEJQWWOoFzayK8y0yNlr_Ew8Bog3kHHNETo1VfArFnpvrnWJUjzh6r_z5I187v0Nv3hNVA**MjUzMjQ3MSxsaW5oYW5nMDIs5p6X6IiqLGxpbmhhbmcwMkBtZWl0dWFuLmNvbSwxLDAzMTExMzgzLDE2Njc0NjUxOTg0Mjc	"
        };

        let url = "http://ones.vip.sankuai.com/api/1.0/ones/projects/9064/issues/search?type=REQUIREMENT"

        var res_1 = request('GET', "www.baidu.com", {});
        // console.log(res_1);
        // var res = request('POST', url, {
        //     json: postData,
        //     headers: headers
        // });

        
        // var user = JSON.parse(res.getBody('utf8'));

        // console.log(user)
    }

    async rp_test_1() {
        let aaa: EntryItem[] = [];
        let that = this;
         let p = await this.rp_test();
       
        console.log(this.res_str);
        console.log("this.res_str");

        //  p.then(function (b:object) {
            
        //     // console.log("rp test");
        //     // console.log(b);
        //     let bb = JSON.stringify(b);
        //     console.log("bb")
        //     console.log(bb)
        //     that.res_str = bb;
        //     // console.log(aaa);
        // });;
    } 
    async rp_test() {
        const postData = JSON.stringify({
            "query":[
                {
                    "field":"state",
                    "type":"TERMS",       
                    "valueList":[
                        "规划中",
                        "开发中",
                        "交互视觉设计",
                        "待开发",
                        "需求评审",
                        "技术评审",
                        "需求设计中"
                    ]
                },
                {
                    "field":"assigned",
                    "type":"TERMS",
                    "valueList":[
                        "muzhantong"
                    ]
                }
            ],
            "displayFieldList" : [
                "id", "name", "projectId", "state"]            
        });
        var options= {
            method:'post',
            uri:"http://ones.vip.sankuai.com/api/1.0/ones/projects/9064/issues/search?type=REQUIREMENT",
            body: postData,
            headers:{
                "ssoClientId":"d2a0562363",
                "ssoId":"eAGFji9IA1EcgHltXJIDwXhxWzh-7_Z7f2bytjmMA4NgkXfv_Q4M3oIbFoMz2PzDTC4ZRJMboqBZg2brwHa6YrBoExUx27_v4yuwqY-HAQvG78P8DiJ_jVY7XZOFRGE7o_UwpdkAHOhIcMGVE4joEiFU2TjuDE94CrI2Zn5piZJFSxlt1hu1WFcbMQjeFHWcj5VqYgW00HWMqjwK3vb6Z_dQZNG_Yf2zNuctPH2eP99C6_H18OISthn3vNjadjfrtDacPz25Os33r_OD3Xy083LcmwxPJqPeDAu2bo5KxV-nzwp_fwMWWomawAhpEmvSqlRkJSUuQqVNmmq-wqWsgAQFKDQsBwRICaYVgrJG4NrgN0vOkjROc3JfwZpvug**eAEFwQkRwEAIBDBLhYE9kMOz-JfQhN5x4-4TcYYjDimKlWfHIdQtv5dLlGqMNTpvrajiZfMDSUASBg**pJV6fyj_WZa4Z_4lYEJQWWOoFzayK8y0yNlr_Ew8Bog3kHHNETo1VfArFnpvrnWJUjzh6r_z5I187v0Nv3hNVA**MjUzMjQ3MSxsaW5oYW5nMDIs5p6X6IiqLGxpbmhhbmcwMkBtZWl0dWFuLmNvbSwxLDAzMTExMzgzLDE2Njc0NjUxOTg0Mjc	"
            },
            json: true
        };
    
        let body = await rp(options);
        console.log(body)
        this.res_str = JSON.stringify(body);
        return this.res_str;
    }

    async test() {
        // await this.sleep();
        console.log("test end");
    }

    // async sleep() {
    //     console.log("enter sleep");
    //     let ret = this.rootItems;
    //     return await new Promise(function(resolve, reject) {
    //         console.log("enter promise");
    //         const req = request("http://ones.vip.sankuai.com/api/1.0/ones/projects/9064/issues/search?type=REQUIREMENT",  {
    //             method: "POST",
    //             headers: {
    //                 "ssoClientId":"d2a0562363",
    //                 "ssoId":"eAGFji9IA1EcgHltXJIDwXhxWzh-7_Z7f2bytjmMA4NgkXfv_Q4M3oIbFoMz2PzDTC4ZRJMboqBZg2brwHa6YrBoExUx27_v4yuwqY-HAQvG78P8DiJ_jVY7XZOFRGE7o_UwpdkAHOhIcMGVE4joEiFU2TjuDE94CrI2Zn5piZJFSxlt1hu1WFcbMQjeFHWcj5VqYgW00HWMqjwK3vb6Z_dQZNG_Yf2zNuctPH2eP99C6_H18OISthn3vNjadjfrtDacPz25Os33r_OD3Xy083LcmwxPJqPeDAu2bo5KxV-nzwp_fwMWWomawAhpEmvSqlRkJSUuQqVNmmq-wqWsgAQFKDQsBwRICaYVgrJG4NrgN0vOkjROc3JfwZpvug**eAEFwQkRwEAIBDBLhYE9kMOz-JfQhN5x4-4TcYYjDimKlWfHIdQtv5dLlGqMNTpvrajiZfMDSUASBg**pJV6fyj_WZa4Z_4lYEJQWWOoFzayK8y0yNlr_Ew8Bog3kHHNETo1VfArFnpvrnWJUjzh6r_z5I187v0Nv3hNVA**MjUzMjQ3MSxsaW5oYW5nMDIs5p6X6IiqLGxpbmhhbmcwMkBtZWl0dWFuLmNvbSwxLDAzMTExMzgzLDE2Njc0NjUxOTg0Mjc	"
    //             }
    //         }, (res)=> {
    //             console.log('enter res');
    //             let rawData = '';
    //             res.on('data', (chunk) => { rawData += chunk; });
    //             res.on('end', () => {
    //                 const parsedData = JSON.parse(rawData);
    
    //                 for (let item of parsedData.data.items) {
    //                     ret.push(new EntryItem(item.name, vscode.TreeItemCollapsibleState.None));
    //                 }

    //                 resolve("success");
    //                 console.log("push to rootItems");
    //                 console.log(ret);
    //                 console.log("push to rootItems done");
    //             });
    //         }
    //         );
    
    //         const postData = JSON.stringify({
    //             "query":[
    //                 {
    //                     "field":"state",
    //                     "type":"TERMS",       
    //                     "valueList":[
    //                         "规划中",
    //                         "开发中",
    //                         "交互视觉设计",
    //                         "待开发",
    //                         "需求评审",
    //                         "技术评审",
    //                         "需求设计中"
    //                     ]
    //                 },
    //                 {
    //                     "field":"assigned",
    //                     "type":"TERMS",
    //                     "valueList":[
    //                         "muzhantong"
    //                     ]
    //                 }
    //             ],
    //             "displayFieldList" : [
    //                 "id", "name", "projectId", "state"]            
    //         });
    //         req.write(postData);
    //         req.end();
    //     });
    //     ;
    // }

    post() {
        const postData = JSON.stringify({
            "query":[
                {
                    "field":"state",
                    "type":"TERMS",       
                    "valueList":[
                        "规划中",
                        "开发中",
                        "交互视觉设计",
                        "待开发",
                        "需求评审",
                        "技术评审",
                        "需求设计中"
                    ]
                },
                {
                    "field":"assigned",
                    "type":"TERMS",
                    "valueList":[
                        "muzhantong"
                    ]
                }
            ],
            "displayFieldList" : [
                "id", "name", "projectId", "state"]            
        });

        let url = 'http://ones.vip.sankuai.com/api/1.0/ones/projects/9064/issues/search?type=REQUIREMENT';
        let headers = {"ssoClientId":"d2a0562363",
                 "ssoId":"eAGFji9IA1EcgHltXJIDwXhxWzh-7_Z7f2bytjmMA4NgkXfv_Q4M3oIbFoMz2PzDTC4ZRJMboqBZg2brwHa6YrBoExUx27_v4yuwqY-HAQvG78P8DiJ_jVY7XZOFRGE7o_UwpdkAHOhIcMGVE4joEiFU2TjuDE94CrI2Zn5piZJFSxlt1hu1WFcbMQjeFHWcj5VqYgW00HWMqjwK3vb6Z_dQZNG_Yf2zNuctPH2eP99C6_H18OISthn3vNjadjfrtDacPz25Os33r_OD3Xy083LcmwxPJqPeDAu2bo5KxV-nzwp_fwMWWomawAhpEmvSqlRkJSUuQqVNmmq-wqWsgAQFKDQsBwRICaYVgrJG4NrgN0vOkjROc3JfwZpvug**eAEFwQkRwEAIBDBLhYE9kMOz-JfQhN5x4-4TcYYjDimKlWfHIdQtv5dLlGqMNTpvrajiZfMDSUASBg**pJV6fyj_WZa4Z_4lYEJQWWOoFzayK8y0yNlr_Ew8Bog3kHHNETo1VfArFnpvrnWJUjzh6r_z5I187v0Nv3hNVA**MjUzMjQ3MSxsaW5oYW5nMDIs5p6X6IiqLGxpbmhhbmcwMkBtZWl0dWFuLmNvbSwxLDAzMTExMzgzLDE2Njc0NjUxOTg0Mjc	"}
        // return axios.post(
        //     'http://ones.vip.sankuai.com/api/1.0/ones/projects/9064/issues/search?type=REQUIREMENT',
        //     postData,
        //     {
        //         headers: {"ssoClientId":"d2a0562363",
        //         "ssoId":"eAGFji9IA1EcgHltXJIDwXhxWzh-7_Z7f2bytjmMA4NgkXfv_Q4M3oIbFoMz2PzDTC4ZRJMboqBZg2brwHa6YrBoExUx27_v4yuwqY-HAQvG78P8DiJ_jVY7XZOFRGE7o_UwpdkAHOhIcMGVE4joEiFU2TjuDE94CrI2Zn5piZJFSxlt1hu1WFcbMQjeFHWcj5VqYgW00HWMqjwK3vb6Z_dQZNG_Yf2zNuctPH2eP99C6_H18OISthn3vNjadjfrtDacPz25Os33r_OD3Xy083LcmwxPJqPeDAu2bo5KxV-nzwp_fwMWWomawAhpEmvSqlRkJSUuQqVNmmq-wqWsgAQFKDQsBwRICaYVgrJG4NrgN0vOkjROc3JfwZpvug**eAEFwQkRwEAIBDBLhYE9kMOz-JfQhN5x4-4TcYYjDimKlWfHIdQtv5dLlGqMNTpvrajiZfMDSUASBg**pJV6fyj_WZa4Z_4lYEJQWWOoFzayK8y0yNlr_Ew8Bog3kHHNETo1VfArFnpvrnWJUjzh6r_z5I187v0Nv3hNVA**MjUzMjQ3MSxsaW5oYW5nMDIs5p6X6IiqLGxpbmhhbmcwMkBtZWl0dWFuLmNvbSwxLDAzMTExMzgzLDE2Njc0NjUxOTg0Mjc	"}
            
                
        //     }
        //   ).then(res => {
        //     console.log("enter post then")
        //     console.log(res.data.items);
        //     this.rootItems = res.data.items;
        //   });

        // return axios({
        //     method: "post",
        //     url,
        //     data: postData,
        //     timeout: 60000,
        //     headers: headers,
        //   })
        //     .then(res => {
        //             console.log("enter post then")
        //             console.log(res.data.items);
        //             this.rootItems = res.data.items;
        //           });

        axios.get
    }
}
